Source/qol/stash.cpp

```diff
@@ -264,11 +264,7 @@ void StartGoldWithdraw()
 
 void WithdrawGold(Player &player, int amount)
 {
-	InitializeItem(player.HoldItem, IDI_GOLD);
-	SetGoldSeed(player, player.HoldItem);
-	player.HoldItem._ivalue = amount;
-	player.HoldItem._iStatFlag = true;
-	ControlSetGoldCurs(player.HoldItem);
+	AddGoldToInventory(player, amount);
 	Stash.gold -= amount;
 	Stash.dirty = true;

```

Comment:
This assumes the player has room for the full amount they requested to withdraw. The return value from `AddGoldToInventory` is ignored, this should be added back to `Stash.gold` or removed from the value that is then subtracted from `Stash.gold`

Reply:
The input is limited to only accept the maximum amount that the user can fit in there inventory so that can never happen.

It's clunky with use of a Global, but that's cause it was copied from how SplitGold was done in the original game :roll_eyes:

Reply:
Ahh, I see. I kinda like the behaviour of allowing the user to attempt to withdraw more than they can hold and leaving the excess behind, it saves having to think when you've got a partially filled inventory since there's no indicator of how much you're allowed to withdraw.

I did find a bug with this implementation though 😄 
![image](https://user-images.githubusercontent.com/357684/159254015-f8a29d2c-b175-4d4e-8d7a-383a86c28768.png)

Reply:
I don't disagree, but also do not want to refactor this with out also improving the other one. Simply adding the check here would possibly cause a lot of work for each key press, and atm I'm looking for the shortest path to a release.